revoke/revoke.go

```diff
@@ -22,14 +24,86 @@ import (
 	"github.com/cloudflare/cfssl/log"
 )
 
-// HardFail determines whether the failure to check the revocation
-// status of a certificate (i.e. due to network failure) causes
-// verification to fail (a hard failure).
-var HardFail = false
+// Revoke type contains configuration for each new revoke instance
+type Revoke struct {
+	// LocalCRL contains path to the local CRL file. When set, certificate
+	// will be checked only using local CRL script, remote methods will be
+	// skipped.
+	localCRL string
+	// HardFail determines whether the failure to check the revocation
+	// status of a certificate (i.e. due to network failure) causes
+	// verification to fail (a hard failure).
+	hardFail bool
+	// crlSet associates a PKIX certificate list with the URL the CRL is
+	// fetched from.
+	crlSet map[string]*pkix.CertificateList
+	// lck is a Mutex locker for the struct
+	lck sync.Mutex
+}
+
+// defaultChecker is a default config for regular apps which don't need to
+// use custom options.
+var defaultChecker = New(false)
+
+// New creates Revoke config structure.
+// Accepts hardfail bool variable as an option
+func New(hardfail bool) *Revoke {
+	return &Revoke{
+		localCRL: "",
+		hardFail: hardfail,
+		crlSet:   map[string]*pkix.CertificateList{},
+	}
+}
+
+// SetLocalCRL sets localCRL path into the Revoke struct
+func (r *Revoke) SetLocalCRL(localCRLpath string) error {
```

Comment:
I think this should go into the `New` function; it's probably not something that should be changed often (ever?) and therefore probably doesn't need its own function.